### ISSUE TITLE
[nit] Calling Flush() before any writing makes a warning

### ISSUE THREAD

**Issue by hakobe** [urfave/negroni#149]
In HandlerFunc, calling `negroni.ResponseWriter.Flush` before calling  `negroni.ResponseWriter.WriteHeader` (or `negroni.ResponseWriter.Write`) makes a warning like below:

```
2016/10/09 14:11:04 http: multiple response.WriteHeader calls
```

This is because `http.ResponseWriter.Flush` calls `http.ResponseWriter.WriteHeader` but `negroni.ResponseWriter.Flush` does not call `negroni.ResponseWriter.WriteHeader` when `negroni.ResponseWriter.Written` is not true.

You can see that warning with this code:

```go
package main

import (
	"fmt"
	"log"
	"net/http"

	"github.com/urfave/negroni"
)

func main() {
	mux := http.NewServeMux()
	mux.HandleFunc("/foo", func(w http.ResponseWriter, r *http.Request) {
		fw := w.(http.Flusher)
		fw.Flush()
		fmt.Fprintf(w, "hello") // http: multiple response.WriteHeader calls
	})

	n := negroni.New()
	n.UseHandler(mux)

	log.Fatal(http.ListenAndServe(":8000", n))
}
```

I think this warning is not matter too much in normal cases :)

**Comment by jszwedko** [comment]
Hi @hakobe,

Thanks for finding that! Definitely a bug and something that should be addressed by #150. Please let me know if that works for you.

**Comment by hakobe** [comment]
Hi @jszwedko,

This fix(#150) works for me. Thanks!

**Comment by jszwedko** [comment]
Awesome, thank you for confirming!
